src/main/java/io/netty/buffer/api/memseg/BifurcatedDrop.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api.memseg;
+
+import io.netty.buffer.api.Drop;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+
+class BifurcatedDrop<T> implements Drop<T> {
+    private static final VarHandle COUNT;
```

Comment:
@chrisvest you use a varHandle to reduce allocations and memory overhead right ?

Reply:
Yeah, don't see a need to have the counter in its own object. `AtomicIntegers` are good if you want to share them across objects.